### PR TITLE
ROX-14763: Verify namespace details refactor

### PR DESCRIPTION
**Pull request by lvalerom**
### Description

Refactor the `Verify namespace details` test.

### User-facing documentation

(*must be* 2 items and both *must be* checked)

- [x] CHANGELOG update is not needed
- [x] Documentation is not needed

### Testing

- [x] inspected CI results

#### Automated testing

(*must be* at least 1 item and all items *must be* checked)

- [x] modified existing tests
  

#### How I validated my change

- [x] Manual test with OCP and GKE clusters
- [x] CI

### CODE REVIEW

File: qa-tests-backend/src/test/groovy/SummaryTest.groovy [hunk 1]
```diff
@@ -117,57 +114,49 @@ class SummaryTest extends BaseSpecification {
...
+        List<String> orchestratorNamespaces = new ArrayList<>()
+        Map<String, String> stackroxNamespaces
+        withRetry(3, 5) {
+            stackroxNamespaces = new HashMap<>()
+            orchestratorNamespaces = orchestrator.getNamespaces()
+
+            NamespaceService.getNamespaces().collect {
+                stackroxNamespaces.put(it.metadata.name, it.metadata.id)
+            }
+            assert stackroxNamespaces.keySet().sort() == orchestratorNamespaces.sort()
```
[the remark this answers is not in the document]
Comment (vikin91) [reply qa-tests-backend/src/test/groovy/SummaryTest.groovy:127]: Here is a code snippet used in different place for that:

```groovy
log.info "In this diff, 'removed' means 'missing in orchestrator but given in ACS', " +
    "whereas 'added' - the other way round"
List<String> stackroxDeploymentNames = Services.getDeployments()*.name
Javers javers = JaversBuilder.javers()
        .withListCompareAlgorithm(ListCompareAlgorithm.AS_SET)
        .build()
log.info javers.compare(stackroxDeploymentNames.sort(), orchestratorResourceNames.sort()).prettyPrint()
```

[the remark this answers is not in the document]
Comment (vikin91) [reply qa-tests-backend/src/test/groovy/SummaryTest.groovy:127]: I see, you use that code later 😅

File: qa-tests-backend/src/test/groovy/SummaryTest.groovy [hunk 2]
```diff
@@ -117,57 +114,49 @@ class SummaryTest extends BaseSpecification {
...
+                            .build()
+                    log.info javers.compare(stackroxDeploymentNames, orchestratorNamespaceDetails.deploymentCount)
+                            .prettyPrint()
+                }
+                assert stackroxNamespaceDetails.numDeployments == orchestratorNamespaceDetails.deploymentCount.size()
+                assert stackroxNamespaceDetails.metadata.clusterId == ClusterService.getClusterId()
+                assert stackroxNamespaceDetails.metadata.name == orchestratorNamespaceDetails.name
+                assert stackroxNamespaceDetails.metadata.labelsMap == orchestratorNamespaceDetails.labels
+                assert stackroxNamespaceDetails.numSecrets == orchestratorNamespaceDetails.secretsCount
+                assert stackroxNamespaceDetails.numNetworkPolicies == orchestratorNamespaceDetails.networkPolicyCount
```
Comment (vikin91) [inline qa-tests-backend/src/test/groovy/SummaryTest.groovy:158]: Will we know the namespace if some of those fail? Maybe a log stating the namespace could be added?

**Review by vikin91 — APPROVED** [review]
Looks good! 
I would proactively add more debug info here and there - I left few optional comments about that.

File: qa-tests-backend/src/test/groovy/SummaryTest.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (lvalerom) [reply qa-tests-backend/src/test/groovy/SummaryTest.groovy:139]: I've added a diff for the namespaces too 👍

File: qa-tests-backend/src/test/groovy/SummaryTest.groovy (hunk 2 above)
Comment (lvalerom) [reply qa-tests-backend/src/test/groovy/SummaryTest.groovy:173]: Makes sense. I've added a log line indicating the namespace we are asserting.

**Pull request merged**